lib/Windowing/Windowing.js

```diff
@@ -1,25 +1,26 @@
-import React, { useState, useCallback, createContext, useEffect } from 'react';
-import { node, number } from 'prop-types';
-import debounce from 'debounce';
+import React, { useState, createContext, useEffect } from 'react';
+import { node, number, arrayOf } from 'prop-types';
 
 export const WindowingContext = createContext({});
 
-const WindowingProvider = ({
+const Windowing = ({
   children,
   containerHeight,
   debounceTimer,
   itemHeight,
-  itemsLength,
   buffer,
-  ...rest
+  allIds
 }) => {
   const [scrollTop, setScrollTop] = useState(null);
   const [height, setHeight] = useState(null);
   const [renderIndexes, setRenderIndexes] = useState({});
+  const [allWindowingIds, setAllWindowingIds] = useState(allIds);
+  const [inViewWindowingIds, setInViewWindowingIds] = useState([]);
 
   const subtractFromStartIndex = index => Math.max(index - buffer, 0);
 
-  const addToEndIndex = index => Math.min(itemsLength, index + buffer);
+  const addToEndIndex = index =>
+    Math.min(allWindowingIds.length, index + buffer);
 
   const getRenderIndexes = () => {
     const startBoundary = scrollTop;
```

Comment:
I feel like we are going a bit of a round about way to calculate the `length` on line 37. I don't think we need to calculate the `endWithBuffer` - this is a hangover from when we used the `endWithBuffer` value in a previous implementation.

To get `length` you can do `(height / itemHeight) + (buffer * 2)`. I think this method makes more sense, are we are no longer calculating the `endWithBuffer` that we don't need.

I'm pretty sure this works, but worth checking!